docs/en/quick-answers/troubleshooting/context-missing-task-step.md

```diff
@@ -1,12 +1,31 @@
 ---
 layout: default
-title: My context is missing the Task/Step but it exists as part of the filename.
+title: Why is my context missing the Task/Step when it exists as part of the filename?
 pagename: context-missing-task-step
 lang: en
 ---
 
-# My context is missing the Task/Step but it exists as part of the filename.
+# Why is my context missing the Task/Step when it exists as part of the filename?
 
-Toolkit uses the path cache to help associate paths on disk with entities in Shotgun and build your context. Since the path cache only stores **folder** information and not filenames, any context-related information contained in the filename (including Task or Step name) will not be detected when building the context.
+When you create folders via Toolkit it registers the path against the entity so that a lookup may be performed. This means that given a path, you can determine the correct context.
+Toolkit will only create registries for folders generated from the schema, and so it doesn't consider things like file names or folders that were defined solely in the `templates.yml`.
+If you don't have a `Task` folder in your schema then you can get in a situation where Toolkit needs to know the task of the file but is unable to work out the task from the path alone.
 
-The best way to solve this currently, is to consider modifying your schema slightly so that the Task or Step is part of the folder structure rather than the filename. 
\ No newline at end of file
+**Example**
+
+Take the default schema structure below; `Asset` and `Step` folders will be registered during the folder creation process:
+
+![Default Asset schema](./images/asset-schema.png)
+
+If you generated a file path using a template like this: `assets/{sg_asset_type}/{Asset}/{Step}/work/maya/{task_name}_{name}.v{version}.{maya_extension}'`, and then attempted to figure out the context from that generated path, it would only be able to establish the `Asset` and the `Step` and **not** the `Task` despite the task's name being in the file path.
+
+**Solution**
+
+Having a `Step` folder and no `Task` folder in your schema is fine for most workflows. Normally you would use the workfiles app to open your scene file by selecting the `Task` you want to work on and then selecting the file. The `Task` you select in the UI is then used to drive the context rather than trying to figure it out from the path of the opened file.
```

Comment:
- Workfiles should be capitalized.
 - Contrary to all my other notes, I feel like "selecting the `Task` you want to work on" should just be "selecting the task you want to work on" – we're not really referring to the entity in Shotgun here, just the name of the task itself, so no need for it to be special. Not married to this; feel free to leave it if you disagree

Reply:
The `Task` vs task was driving me mad. I'll follow your suggestion!